File: ffi/python/blsct/child_key.py
import blsct
from typing import Any
from .scalar_based_key import ScalarBasedKey
from ..managed_obj import ManagedObj
from ..scalar import Scalar
from child_key_derived.blinding_key import BlindingKey
from child_key_derived.token_key import TokenKey
from child_key_derived.tx_key import TxKey

class ChildKey(ScalarBasedKey):
  def __init__(self, seed: Scalar):
    # child_key is of type *BlsctScalar
    child_key = blsct.from_seed_to_child_key(seed.value())
    super().__init__(child_key)

  def to_blinding_key(self) -> BlindingKey:
    return BlindingKey(self)

  def to_token_key(self) -> TokenKey:
    return TokenKey(self)

  def to_tx_key(self) -> TxKey:
    return TxKey(self)

File: ffi/ts/src/__tests__/keys/childKey.ts
import { Scalar } from '../../scalar'
import { ChildKey } from '../../keys/childKey'

describe('KeyDerivation', () => {
  test('derive from childKey', () => {
    const seed = new Scalar(12345)
    const childKey = new ChildKey(seed)
  })

  test('toBlindingKey', () => {
    const seed = new Scalar(12345)
    const x = new ChildKey(seed)
    x.toBlindingKey()
  })

  test('toTokenKey', () => {
    const seed = new Scalar(12345)
    const x = new ChildKey(seed)
    x.toTokenKey()
  })

  test('toTxKey', () => {
    const seed = new Scalar(12345)
    const x = new ChildKey(seed)
    x.toTxKey()
  })
})
